Makefile

```diff
@@ -33,6 +33,9 @@ precommit: $(TOOLS_DIR)/golangci-lint  $(TOOLS_DIR)/misspell $(TOOLS_DIR)/string
 	PATH="$(abspath $(TOOLS_DIR)):$${PATH}" go generate ./...
 	$(TOOLS_DIR)/golangci-lint run --fix # TODO: Fix this on windows.
 	$(TOOLS_DIR)/misspell -w $(ALL_DOCS)
+
+.PHONY: mod-tidy
```

Comment:
I'm kinda on the fence with this. I guess it's fine as long as you make `mod-tidy` a prerequisite of the `precommit` target. That way CI can run `go mod tidy` and yell at you when the `go.mod` or `go.sum` files change.